File: data/gpt-4o/readable_format_HPEC/cpp/file220.cpp
void init_array(int length, int **c, double **w){
 for(int i=0; i<length; i++)
 for(int j=0; j<length; j++){
 c[j][i] = ((i)*(j)) % 2;
 w[j][i] = (double) ((i) - (j)) / (double) length;
 }
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file220.f90
subroutine init_array(length, c, w)
implicit none

DATA_TYPE, dimension(length, length) :: w, c
integer :: i, j
integer length

do i = 1, length
do j = 1, length
c(j, i) = mod((i-1)*(j-1), 2)
w(j, i) = (DBLE((i - 1) - (j - 1))) / DBLE(length)
end do
end do
end subroutine
